src/services/partner.ts

```diff
@@ -0,0 +1,62 @@
+import { CallOverrides } from "@ethersproject/contracts";
+
+import { ChainId } from "../chain";
+import { ContractService } from "../common";
+import { Context } from "../context";
+
+const YVECRV = "0xc5bDdf9843308380375a611c18B50Fb9341f502A";
+const PSLPYVBOOSTETH = "0xCeD67a187b923F0E5ebcc77C7f2F7da20099e378";
+
+interface PartnerServiceParams<T extends ChainId> {
+  chainId: T;
+  ctx: Context;
+  address: string;
+}
+/**
+ * [[PartnerService]] provides access to yearns partner contract.
+ * It's implemented in the form of a contract that lives on all networks
+ * supported by yearn.
+ */
+export class PartnerService<T extends ChainId> extends ContractService<T> {
+  static abi = ["function deposit(address vault, address partnerId, uint256 amount) external returns (uint256)"];
+  static IGNORED_ADDRESSES = [YVECRV, PSLPYVBOOSTETH];
+  partnerId: string;
+
+  constructor({ chainId, ctx, address }: PartnerServiceParams<T>) {
+    super(address, chainId, ctx);
+    this.partnerId = ctx.partnerId;
+  }
+
+  /**
+   * Get most up-to-date address of the partner contract for a particular chain id.
+   * @param chainId
+   * @returns address
+   */
+  static addressByChain(chainId: ChainId): string | null {
+    switch (chainId) {
+      case 1:
+        return "0x8ee392a4787397126C163Cb9844d7c447da419D8";
+      case 250:
+      case 1337:
+      case 42161:
+      default:
+        return null;
+    }
+  }
+
+  isAllowed(vault: string) {
```

Comment:
We could in the future make the `vault` a bit more type safe with its own `VaultType`

No action needed now, just food for thought